src/app/api/github/route.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * POST - Link userId and github username to be verified later
+ * PUT - Verify the userId and github from db
+ */
+
+import { NextRequest, NextResponse } from 'next/server';
+import db from '@/db';
+import { getServerSession } from 'next-auth';
+import { authOptions } from '@/lib/auth';
+
+export async function POST(req: NextRequest) {
+  const session = await getServerSession(authOptions);
+
+  if (!session || !session?.user) {
+    return NextResponse.json(
+      { message: 'Authentication failed' },
+      { status: 401 },
+    );
+  }
+
+  const { value: username, userId } = await req.json();
```

Comment:
You can use Zod to check if data has correct type that will prevent the inconsistent data in DB.

Reply:
added it thanks